store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/impl/FederatedOperationHandler.java

```diff
@@ -97,13 +105,23 @@ private Iterable getAllGraphResults(final FederatedOperation<INPUT, OUTPUT> oper
 
     }
 
-    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation operation, final FederatedStore store) throws OperationException {
+    private Object mergeResults(final Iterable resultsFromAllGraphs, final FederatedOperation<INPUT, OUTPUT> operation, final FederatedStore store) throws OperationException {
         try {
             Object rtn = null;
 
-            final BiFunction mergeFunction = getMergeFunction(operation, store, context, isEmpty(resultsFromAllGraphs));
+            BiFunction mergeFunction = getMergeFunction(operation, store, context, isEmpty(resultsFromAllGraphs));
+            final List<GraphSerialisable> graphs = getGraphs(operation, context, store);
+
+            // If default merging and only have one graph or no common groups then just return the current results
+            if (!graphs.isEmpty()
+                    && mergeFunction instanceof ApplyViewToElementsFunction
+                    && (graphs.size() == 1 || !graphsHaveCommonSchemaGroups(graphs))) {
+                LOGGER.info("Returning flat list of results as complex merging not required when only one graph or no common groups");
+                // Just use the concatenate merge to flatten the results
+                mergeFunction = new ConcatenateMergeFunction();
```

Comment:
This could maybe be tested to improve coverage?

Reply:
Coverage should hopefully be sorted now, there was an issue with how it was getting the graphs which is now fixed so existing tests appear to cover this functionality.